### PR TITLE
Refactor kmPlot issue #199

### PR DESCRIPTION
**Pull request by syt123450**
In this PR, I refactor the kmPlot, remove the bootstrap dependencies, and adjust some UI.

In remove bootstrap dependencies part:
1. Replace the bootstrap widget with react-toolbox widget
2. Replace the bootstrap classes with our own classes, defined in "km.css".

In adjust UI part:
1. Align the PDF button
2. Remove the close button(I think there is a close icon in the right top corner, there is no need to have one special close button)
3. Add icon in PDF button and Help button to let them more representative
4. Change the help link with help button, so there is a uniformed UI
5. Change the cursor style (when user hover on the right list, the original cursor style is "text", is as if the list is editable, it is a little bit puzzling, I change it with the default cursor)
6. Adjust the basic UI including margin, padding, and so on.

PS. All function has been tested, they can work well in Chrome, Safari, and Firefox.

The original kmPlot UI is as shown below:
kmplot
pwarning

The kmPlot UI after refactoring is as shown below:
kmplotnew
pwarningnew

### CODE REVIEW

**Comment by acthp** [comment]
@syt123450 Cool, looks pretty good. There are again some unrelated indent changes which should be removed from the diff. A bigger issue is that the root of the modal has changed, such that it covers the navigation menu. We definitely need the navigation menu to remain usable while the km is open. The overflow behavior has changed, as well, such that the pop-up has a scroll bar, instead of the page.

I recommend reviewing the css in production vs. the diff, in the debugger, and especially around which items are positioned, and how. Ping me if you get stuck.

**Comment by syt123450** [comment]
@acthp I dig into the two issue you mentioned above, I fix one and we may need to discuss for another.

The UI now look like this:
kmplotupdate


The one I have fixed is the navbar. The navbar now will not be hidden by overlay (the "curtain" under the dialog). My solution is set navbar "position:relative; z-index: 201;"(201 is the min number I found which can be above the overlay). As I am afraid that change may be influence other page, I tested other pages, it looks good in my test case. If this influences somewhere I have not found, I will fix it.

The other one is the overflow different with previous one. The "modal" we used in bootstrap is designed to be rendered in a div, so we can set the "position" of parent node to let the modal to be render in a specific div. However, I find that the React-toolbox seems to have a different design concept. It will always append the dialog onto the "root", whatever position I set to its parent div, it won't be render onto its parent, even if I set the "position: absolute;". I have began to consider this issue since I began to refactor the modal with dialog, however, I have not find the solution on the website yet.

So as a workaround, I change some UI of kmPlot. Now the kmPlot can be render in "one page" without scroll bar in it, it may give user a feeling that both pop-up and the page can not be scroll, only if they close the pop-up, can they scroll the page. I think this is also a conventional UX design. If it is important to render the pop-up into a specific div, I am afraid we may need to find other library to draw this pop-up.

If these changes are ok, I will merge all these commits into one.

Thanks.

**Comment by acthp** [comment]
@syt123450 Is your latest code on this branch? I pulled d3f2b9f9, but it doesn't look like your screenshot. I'm happy to look at your current workaround, if you push it to github.

I looked at the react-toolbox Dialog code. It's built on a Portal component, which appears to allow embedding in a container, but the container property is not exposed by Dialog.

I'm not sure we actually need Portal, since it's purpose is to punch through the DOM hierarchy. We already render KM in the top Application div, where it belongs. Perhaps KmPlot just needs to be horizontally centered, with an absolutely positioned overlay to dim the spreadsheet, and a close button?

**Comment by syt123450** [comment]
@acthp I think I have been push all my code on this branch, can you give my a screen shot? So I can fix it. I am afraid its a responsive problem, It look good in my 15 inch screen, I may fix it to fit for different size of screens.

**Comment by acthp** [comment]
@syt123450 Oh, I see now. When I opened KM the page was scrolled down a bit. The workaround prevents scrolling, so I couldn't access the nav bar. If I start at the top of the page, the nav bar is accessible. I don't think we want it to be like this. On my screen I'm also getting a drop shadow from the nav bar on the AppControls component, which doesn't look good.

What do you think of trying it without Dialog, just using absolute positioning? Something like

```
<div className='kmOverlay'>
   <Card>
       {Content}
   </Card>
</div>
```
where kmOverlay is styled to cover Application, and Card is used to render a title and drop shadow?

**Comment by acthp** [comment]
@syt123450 I'm not enthusiastic about the z-index work-around, because of the shadow problem. The nav needs to be below AppControls, but the workaround requires nav above AppControls. The result is a shadow from nav on AppControls:

shadow

I looked into the react-toolkit code a bit more, and found that the overlay and wrapper can be themed via the ```wrapper``` and ```overlay``` theme keys, like

```
<Dialog
    theme={{
        wrapper: 'kmWrapper',
        overlay: 'kmOverlay'
   }}
```

They can then be positioned below the nav, like

```
.kmWrapper {
	display: block;
	position: absolute;
	height: 100%;
	top: 64px; /*var(--appbar-height);*/
}

.kmOverlay {
	position: absolute;
	height: 100%;
	top: 0;
}
```

The ```display: block``` is working around a different issue. In the PR the height is now controlled in two different places: the ```dims``` property (see ```getDefaultProps```), and the new ```height``` setting on ```.kmDialog```. They have to be updated simultaneously in order to render correctly, which is bad. I dropped the new ```height``` setting, but then needed to change ```display``` in order for the height to shrink to the content size. Final css looks like

```
div.kmDialog {
	width: 850px;
	overflow: auto;
	padding-top: 10px;
	margin: 20px auto 0 auto;
}
```

Also, regarding this rule, I note you've been adding tags to the classes, like ```div.kmDialog``` instead of ```.kmDialog```. What's the point of add the tag? I don't think we need it. Same for ```button.kmDialogClose```, etc.

Can you make these changes, drop ```.NavAppBar```, and remove whitespace from the diff?

**Comment by syt123450** [comment]
@acthp Thank you so much for your guidance, the setting of the theme is really a break through for me. Based on your sample code, I did the changes below:

1. Adjust the theme of kmDialog and warningDialog
2. Shows the footer of the page, the footer now will not be shadowed by overlay, the effect is as the screen shot below:
footer
3. Remove css class .NavAppBar
4. I remove some of the css definition like "div.kmDialog", however, I still keep the "button.kmDialogClose" unchanged, as only ".kmDialogClose" can not influence the button's background.

**Comment by acthp** [comment]
This looks great! There's still a lot of whitespace change, which makes the diff larger & harder to review. If you're having trouble eliminating your whitespace changes, drop me an email & we'll walk through it. I'll tag some of the lines in the diff so you know what I'm talking about.

Changing whitespace in a line you're editing for the diff is fine. Changing it in unrelated lines creates noise in the diff. If whitespace needs fixing, it's best addressed in a separate PR, with commit log "whitespace" or "fixing whitespace".

Can you back out whitespace changes, and rebase to a single commit for the PR?

File: js/KmPlot.js [hunk 1]
```diff
@@ -28,11 +30,11 @@ function censorLines(xScale, yScale, censors, className) {
 			key={i}
 			className={className}
 			x1={0} x2={0} y1={-5} y2={5}
-			transform={`translate(${xScale(t)},${yScale(s)})`} />
+			transform={`translate(${xScale(t)},${yScale(s)})`}/>
```
Comment (acthp) [inline js/KmPlot.js]: ws

File: js/KmPlot.js [hunk 2]
```diff
@@ -234,10 +258,10 @@ var Legend = React.createClass({
 	},
 
 	render: function () {
-		let { groups, setActiveLabel, activeLabel } = this.props;
-		let { colors, curves, labels } = groups;
+		let {groups, setActiveLabel, activeLabel} = this.props;
```
Comment (acthp) [inline js/KmPlot.js]: ws

File: js/KmPlot.js [hunk 3]
```diff
@@ -234,10 +258,10 @@ var Legend = React.createClass({
...
+		let {colors, curves, labels} = groups;
 		let sets = _.zip(colors, curves, labels)
-					.map(set => makeLegendKey(set, setActiveLabel, activeLabel));
+			.map(set => makeLegendKey(set, setActiveLabel, activeLabel));
```
Comment (acthp) [inline js/KmPlot.js]: ws

File: js/KmPlot.js [hunk 4]
```diff
@@ -312,8 +338,8 @@ var KmPlot = React.createClass({
 		}
 	}),
 
-	getInitialState: function() {
-		return { activeLabel: '' };
+	getInitialState: function () {
```
Comment (acthp) [inline js/KmPlot.js]: ws

File: js/nav.module.css [hunk 5]
```diff
@@ -9,4 +9,4 @@
 .logoXena {
 	max-width: 139px;
 	height: auto;
-}
+}
```
Comment (acthp) [inline js/nav.module.css]: ws

**Comment by syt123450** [comment]
@acthp Sorry for misunderstanding the whitespace issue, I will fix it right now.

**Pull request merged**